app/models/user.rb

```diff
@@ -6,6 +6,27 @@ class User < ApplicationRecord
   has_many :replies, dependent: :destroy
   has_many :requests, through: :replies
 
+  def self.upsert_via_telegram(message)
+    from, chat = message.values_at('from', 'chat')
+    telegram_chat_id = chat['id']
+    telegram_id, username, first_name, last_name = from.values_at('id', 'username', 'first_name', 'last_name')
+    user = User.find_by(telegram_id: telegram_id)
```

Comment:
You should be able to use [find_or_initialize_by](https://edgeapi.rubyonrails.org/classes/ActiveRecord/Relation.html#method-i-find_or_initialize_by):

```ruby
User
  .find_or_initialize_by(telegram_id: telegram_id)
  .update(
    username: username,
    telegram_chat_id: telegram_chat_id,
    username: username,
    first_name: first_name,
    last_name: last_name
  )
...
```

Reply:
We don't want `first_name` or `last_name` to be updated. `update` will not go through validations.